Android-Push-Notifications/Controllers/DevicesController.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Entity;
+using System.Linq;
+using System.Net;
+using System.Web;
+using System.Web.Mvc;
+using Android_Push_Notifications.Models;
+
+namespace Android_Push_Notifications.Controllers
+{
+    public class DevicesController : Controller
+    {
+        private DbContext db = new DbContext();
```

Comment:
move to Repositories all db operations.
use Services to call Repositories methods